### DISCUSSION TITLE
How do you specify a custom dotenv file to load instead of the default .env

### DISCUSSION THREAD

**Discussion by iongion** [dagster-io/dagster#18260] [Q&A] [answered]
Trying to add dagster to an existing project, where I have this pattern

- `.env` - This is with default values and no secrets - present in git
- `.env.local` - This is with local overrides - ignored from git, can contain secrets

How can I specify dagster to use `.env.local` instead ?

**Comment by gibsondan** [comment, marked as answer]
We have the ability to customize the folder that's being used to find the .env file, but we don't currently have the ability to customize the file name - it is assumed to be ".env" here: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/secrets/env_file.py#L33

However one workaround would be to load from the .env file yourself in Python as a side effect in your code when your definitions are being loaded. Python packages like the python-dotenv package can help with this: https://pypi.org/project/python-dotenv/
